### PR TITLE
fixed readme and added purge of authorized_keys file

### PR DESCRIPTION
**Pull request by slimakcz**

### CODE REVIEW

**Comment by cloudlab-autobuild** [comment]
Can one of the admins verify this patch?

File: openssh/server/public_key.sls [hunk 1]
```diff
@@ -39,6 +39,14 @@
 
 {%- endif %}
 
+{%- else %}
+
+{%- if user.get('purge', False) %}
```
Comment (fpytloun) [inline openssh/server/public_key.sls]: Convention is to used `enabled` keyword which if is false instructs states to remove given resources.

**Comment by slimakcz** [comment]
Is it better now?

**Comment by fpytloun** [comment]
I am looking at it a little bit deeper and seems that I misunderstood your initial idea.

If you have purge set to true, whole authorized_keys file is enforced, otherwise only new keys are added.

If purge is set to true and no public_keys are defined, empty file should be created - you want to remove that file completely in such case, right?

Then I would suggest using this:
```
{%- if user.get('public_keys', []) %}
file.managed as it's in place inside purge=True conditional
{%- else %}
your file.absent
{%- endif %}
```

File: README.rst [hunk 2]
```diff
@@ -36,6 +36,7 @@ OpenSSH client with individual private key and known host
           root:
             enabled: true
             user: ${linux:system:user:root}
+            home: ${linux:system:user:root:home}
```
Comment (fpytloun) [inline README.rst]: This will have no effect as user.home is referenced

**Comment by slimakcz** [comment]
Not exactly, because in your original code if you have empty list of public_keys, then authorized_keys isn't empty. In this case authorized_keys file isn't changed and still have old keys, because you have first check of public_keys and next step is purge check. Empty file will be ok, but isn't in your code.
Line 8,10 from your original code:
{%- if user.public_keys is defined %}
{%- if user.get('purge', False) %}

File: README.rst (hunk 2 above)
Comment (slimakcz) [inline README.rst]: ok, sorry. my mistake

**Comment by fpytloun** [comment]
Got it, so first version was probably correct - if public_keys are not defined and purge is set to True, then ensure file doesn't exist.

**Comment by slimakcz** [comment]
Ok, and what now? It will be possible approve / add to upstream first version?

**Comment by fpytloun** [comment]
Please update this pull req to match correct state + fixes my comments (like the readme, etc.) so I can merge it.

**Pull request closed without merging**